pallets/liquid-staking/src/lib.rs

```diff
@@ -1732,6 +1765,75 @@ pub mod pallet {
             Ok(())
         }
 
+        #[require_transactional]
+        fn do_fast_match_unstake(unstaker: &T::AccountId) -> DispatchResult {
+            FastUnstakeRequests::<T>::try_mutate_exists(unstaker, |b| -> DispatchResult {
+                if let Some(request_amount) = b.take() {
+                    let available_amount = Self::matching_pool().total_stake_amount.free()?;
+                    let actual_amount = available_amount.min(request_amount);
+
+                    if !actual_amount.is_zero() {
+                        let fee = T::MatchingPoolFastUnstakeFee::get()
+                            .checked_mul_int(actual_amount)
+                            .ok_or(ArithmeticError::Overflow)?;
+                        let transfer_amount = actual_amount
+                            .checked_sub(fee)
+                            .ok_or(ArithmeticError::Underflow)?;
+
+                        // decrease storage `Unlockings`
+                        Unlockings::<T>::try_mutate(unstaker, |b| -> DispatchResult {
+                            let chunks = b.as_mut().ok_or(Error::<T>::NoUnlockings)?;
+                            let mut matched_balance: Balance = Zero::zero();
+
+                            while let Some(last) = chunks.last_mut() {
+                                if matched_balance + last.value <= actual_amount {
+                                    matched_balance += last.value;
+                                    chunks.pop();
+                                } else {
+                                    let diff = actual_amount - matched_balance;
+
+                                    matched_balance += diff;
+                                    last.value -= diff;
+                                }
+                                if matched_balance >= actual_amount {
+                                    break;
+                                }
+                            }
+
+                            Ok(())
+                        })?;
+
+                        // decrease storage `MatchingPool`
+                        MatchingPool::<T>::try_mutate(|p| -> DispatchResult {
+                            p.sub_stake_amount(actual_amount)?;
```

Comment:
we can support another mode which is not matching against the pool but matching against the `fast_match_unstake` caller.

this means:

- we dont `sub_stake_amount` from the pool
- we substract the caller's balance instead
- `fast_match_unstake` will need to add another parameter `stake_amount`

this looks like `matcher` can also stake at the same time with some extra privileges for them to do this. Like:

- we probably dont need to charge staking_fee anymore because this is just sDOT exchange, no new sDOT minted thus dont share rewards

Reply:
@MrPai how do you think about this? 

Though we can finish first the other parts and then review this again later

Reply:
This could be an optional way, and it should be decided by FE how to do the match.